### PR TITLE
Issue 42768: Add support for assigning DOIs to datasets submitted to Panorama Public

### PR DESCRIPTION
**Pull request by vagisha**
#### Rationale
We are partnering with UW Libraries to assign Digital Object Identifiers (DOIs) to datasets submitted to Panorama Public. UW Libraries is a member of DataCite, a DOI registration organization primarily for research datasets. 

#### Changes
* Admins can enter the DataCite credentials through the Panorama Public Admin Console link
* A DOI is assigned to a dataset when it is copied to Panorama Public.  The admin doing the copy can decide if a DOI is assigned or not with a checkbox in the copy form
* DOI can also be assigned, deleted or published from the NavMenu added to the TargetedMSExperimentWebPart

### CODE REVIEW

File: panoramapublic/src/org/labkey/panoramapublic/PanoramaPublicController.java [hunk 1]
```diff
@@ -705,6 +716,176 @@ public Container getSupportContainer()
...
+        @Override
+        public ModelAndView getView(DataCiteCredentialsForm form, boolean reshow, BindException errors)
+        {
+            if(!reshow)
+            {
+                PropertyManager.PropertyMap map = PropertyManager.getEncryptedStore().getWritableProperties(DataCiteService.CREDENTIALS, false);
+                if(map != null)
+                {
+                    form.setUser(map.get(DataCiteService.USER));
+                    form.setPassword(map.get(DataCiteService.PASSWORD));
```
Comment (labkey-jeckels) [inline panoramapublic/src/org/labkey/panoramapublic/PanoramaPublicController.java]: Up to you, but we've gotten dinged on security reviews for echoing a previously entered password back into the HTML for forms like this. Instead, we've switched to forcing users to always reenter passwords when submitting the form.

File: panoramapublic/src/org/labkey/panoramapublic/datacite/DataCiteException.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        super("Request failed." + message + ". Code: " + response.getResponseCode() + "; Message " + response.getMessage() + "; Body: " + response.getResponseBody());
+        _response = response;
+    }
+
+    public HtmlString getHtmlString()
+    {
+        if(_response != null)
+        {
+            String s = "<div>Code: " + _response.getResponseCode() + " <br/> Message: " + _response.getMessage() + " <br/> " + _response.getResponseBody();
```
Comment (labkey-jeckels) [inline panoramapublic/src/org/labkey/panoramapublic/datacite/DataCiteException.java]: Does the message here contain trusted HTML? If not, can it be encoded?

**Review by labkey-jeckels — APPROVED** [review]

File: panoramapublic/src/org/labkey/panoramapublic/datacite/DataCiteException.java (hunk 2 above)
Comment (vagisha) [reply panoramapublic/src/org/labkey/panoramapublic/datacite/DataCiteException.java]: Done. It should have been encoded. Thanks.

File: panoramapublic/src/org/labkey/panoramapublic/PanoramaPublicController.java (hunk 1 above)
Comment (vagisha) [reply panoramapublic/src/org/labkey/panoramapublic/PanoramaPublicController.java]: Done.

**Pull request merged**